### PR TITLE
add tasks

### PR DESCRIPTION
**Pull request by HiQoSolutionsAI**

### CODE REVIEW

File: src/app/board/list/list.component.scss [hunk 1]
```diff
@@ -24,3 +24,7 @@
 .mat-input-element {
   caret-color: $color-basic;
 }
+
+.list-content {
+  margin-bottom: 0;
+}
```
Comment (EkaterinaSaverina) [inline src/app/board/list/list.component.scss]: no need to create this class
you can use this property from utils

File: src/app/board/list/list.component.ts [hunk 2]
```diff
@@ -39,4 +59,52 @@ export class ListComponent {
...
+
+    this.tasks$ = await this.listService.getTasks(listId);
+  }
+
+  async deleteTask({ listId, id }: Partial<Task>): Promise<void> {
+    const result = this.dialog.openConfirmationDialog({
+      typeDialog: DialogModalComponent,
+      message: deleteMessage,
+    });
+    if (await firstValueFrom(result)) {
```
Comment (EkaterinaSaverina) [inline src/app/board/list/list.component.ts]: it's better to move this condition to variable (const)

File: src/app/board/task/task.component.scss [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+.task-wrapper {
+  padding: 20px 10px;
+  border-bottom: solid 1px $color-basic;
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  box-sizing: border-box;
+  cursor: move;
+  font-size: 14px;
```
Comment (EkaterinaSaverina) [inline src/app/board/task/task.component.scss]: use class from typography

File: src/app/board/task/task.component.html [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<section class="tasks-wrapper">
+  <div class="task-wrapper">
```
Comment (EkaterinaSaverina) [inline src/app/board/task/task.component.html]: do you need to use this class for both tags?

File: src/app/board/task/task.component.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  styleUrls: ['./task.component.scss'],
+})
+export class TaskComponent {
+  @Input() listId!: string;
+  @Input() taskId!: string;
+  @Input() taskTitle!: string;
+
+  @Output() deleteTask = new EventEmitter<Partial<Task>>();
+
+  constructor() {}
```
Comment (EkaterinaSaverina) [inline src/app/board/task/task.component.ts]: empty constructor :D

File: src/app/core/services/boards.service.ts [hunk 6]
```diff
@@ -15,25 +15,23 @@ export class BoardsService {
...
 
-    this.sourceLists.next(await firstValueFrom(result as Observable<List[]>));
+    this.sourceLists.next(result);
   }
 
   async updateListTitle(listId: string, titleValue: string | undefined): Promise<void> {
-    const result = this.lists.pipe(
+    const updatedLists = this.lists.pipe(
       map((array) =>
         array.map(function (element) {
```
Comment (EkaterinaSaverina) [inline src/app/core/services/boards.service.ts]: please use arrow function

File: src/app/core/services/list.service.ts [hunk 7]
```diff
@@ -0,0 +1,57 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, firstValueFrom, map, Observable } from 'rxjs';
+
+import { HttpService } from '@app/core/services';
+import { Task } from '@app/core/models';
+
+import { groupBy } from 'lodash';
```
Comment (EkaterinaSaverina) [inline src/app/core/services/list.service.ts]: move this import to the core ones

File: src/app/board/task/task.component.html (hunk 4 above)
Comment (HiQoSolutionsAI) [reply src/app/board/task/task.component.html]: these are different classes)

**Review by EkaterinaSaverina — APPROVED** [review]

**Pull request merged**